src/librbd/ImageCtx.cc

```diff
@@ -882,20 +879,22 @@ class ThreadPoolSingleton : public ThreadPool {
                                                  pairs, &res);
       for (map<string, bufferlist>::iterator it = res.begin();
            it != res.end(); ++it) {
-        j = local_config_t.set_val(it->first.c_str(), it->second.c_str());
+        string val(it->second.c_str(), it->second.length());
+        j = local_config_t.set_val(it->first.c_str(), val);
         if (j < 0) {
           lderr(cct) << __func__ << " failed to set config " << it->first
                      << " with value " << it->second.c_str() << ": " << j
                      << dendl;
         }
-        break;
       }
       start = pairs.rbegin()->first;
     }
 
 #define ASSIGN_OPTION(config)                                                  \
     do {                                                                       \
-      if (configs[#config])                                                    \
+      string key = "rbd_";						       \
```

Comment:
i think we need this rbd_ prefix since all the keys in configs are prefixed by rbd_